compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -611,7 +611,7 @@ public void visit(BLangClassDefinition classDefinition) {
             objectType = isReadOnly ? new BObjectType(tSymbol, Flags.READONLY) : new BObjectType(tSymbol);
         }
 
-        if (flags.contains(Flag.DISTINCT)) {
+        if (flags.contains(Flag.DISTINCT) || classDefinition.induceTypeIds) {
```

Comment:
With this[1] change, I don't think we need this check and this flag

[1] - https://github.com/ballerina-platform/ballerina-lang/pull/26098/files#diff-fa94fe90a875c08eaf806f89f6ceb7eec7205acaf75c88a41e3f7103abfa5a52L410

Reply:
Need to set the `typeIdSet` for `class` created for object ctor so that it can be accessed in `populateDistinctTypeIdsFromIncludedTypeReferences`. 
Now it is not the default unmodifiable list.

Reply:
Done